torch/distributed/fsdp/_wrap_utils.py

```diff
@@ -119,3 +120,98 @@ def _warn_on_overridden_mixed_precision(
         "These modules will be wrapped as separate FSDP instacnes with mixed "
         "precision disabled."
     )
+
+
+def _validate_frozen_params(
+    root_module: nn.Module,
+    modules_to_wrap: Set[nn.Module],
```

Comment:
This PR still needs to be connected to the auto wrapping changes in previous PRs in the stack, but the point is that if we know `modules_to_wrap` in advance, then we can provide a better error message and include this check as part of auto wrapping.